code/game/objects/structures/morgue.dm

```diff
@@ -80,22 +80,14 @@
 	if(Adjacent(user))
 		return attack_hand(user)
 
-/obj/structure/morgue/attackby(P, mob/user)
-	if(istype(P, /obj/item/pen))
-		var/new_label = sanitize_safe(input(user, "What would you like the label to be?", capitalize(name), null) as text|null, MAX_NAME_LEN)
-
-		if((!Adjacent(user) || loc == user))
+/obj/structure/morgue/attackby(obj/item/P, mob/user)
+	if(IS_PEN(P) && CanPhysicallyInteractWith(user, src) && loc != user)
```

Comment:
I feel like in an ideal world we could move this attackby checking down to the base level and have it check for the presence of the extension, call generic procs, etc.

Reply:
Honestly, the way this thing uses label is non-standard. We could probably just flush that stuff here, and let the hand labeler do it. 
I wouldn't move the pen interaction to a base class because a lot of things just use pens to rename themselves in some cases and in some others to do other things.  Ideally, I feel we probably should phase out using the pen on things to rename them. Machines could do it via multitool interactions or something, and the rest could be done through some alt extension?

Reply:
Anyways, I just removed labeling by pen on this and the crematorium. So now its using the standard way with the labeler.